source/server/http/admin.cc

```diff
@@ -529,11 +545,37 @@ Http::Code AdminImpl::handlerClusters(absl::string_view url, Http::HeaderMap& re
   return Http::Code::OK;
 }
 
-// TODO(jsedgwick) Use query params to list available dumps, selectively dump, etc
-Http::Code AdminImpl::handlerConfigDump(absl::string_view, Http::HeaderMap& response_headers,
+void AdminImpl::writeDumpKeysAsJson(Envoy::Buffer::Instance& response) const {
+  envoy::admin::v2alpha::ConfigDumpKeys keys;
+  for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
+    keys.add_keys(key_callback_pair.first);
+  }
+  response.add(MessageUtil::getJsonStringFromMessage(keys, true)); // pretty-print
+}
+
+Http::Code AdminImpl::handlerConfigDump(absl::string_view url, Http::HeaderMap& response_headers,
                                         Buffer::Instance& response, AdminStream&) const {
+  Http::Utility::QueryParams query_params = Http::Utility::parseQueryString(url);
+  if (hasListKeysParam(query_params)) {
+    writeDumpKeysAsJson(response);
+    return Http::Code::OK;
+  }
+
+  auto callbacks_map = config_tracker_.getCallbacksMap();
```

Comment:
This is sub optimal from performance standpoint as it will make deep copy of the callbacks map for each admin request. It maybe ok for the admin interface, if the expected rate of config_dump requests is low.

Reply:
Ah, cool catch. I only did that to remove some messy branching. I'll keep this in mind when I draft the change suggested above.